src/main/java/org/kohsuke/github/GHRepositoryRule.java

```diff
@@ -0,0 +1,692 @@
+                "required_approving_review_count");
+        /**
+         * required_review_thread_resolution parameter
+         */
+        public static final BooleanParameter REQUIRED_REVIEW_THREAD_RESOLUTION = new BooleanParameter(
+                "required_review_thread_resolution");
+        /**
+         * required_status_checks parameter
+         */
+        public static final ListParameter<StatusCheckConfiguration> REQUIRED_STATUS_CHECKS = new ListParameter<StatusCheckConfiguration>(
+                "required_status_checks") {
+            @Override
+            TypeReference<List<StatusCheckConfiguration>> getType() {
+                return new TypeReference<List<StatusCheckConfiguration>>() {
+                };
+            }
+        };
+        /**
+         * strict_required_status_checks_policy parameter
+         */
+        public static final BooleanParameter STRICT_REQUIRED_STATUS_CHECKS_POLICY = new BooleanParameter(
+                "strict_required_status_checks_policy");
+        /**
+         * name parameter
+         */
+        public static final StringParameter NAME = new StringParameter("name");
+        /**
+         * negate parameter
+         */
+        public static final BooleanParameter NEGATE = new BooleanParameter("negate");
+        /**
+         * operator parameter
+         */
+        public static final Parameter<Operator> OPERATOR = new Parameter<Operator>("operator") {
+            @Override
+            TypeReference<Operator> getType() {
+                return new TypeReference<Operator>() {
+                };
+            }
+        };
+        /**
+         * regex parameter
+         */
+        public static final StringParameter REGEX = new StringParameter("regex");
+        /**
+         * workflows parameter
+         */
+        public static final ListParameter<WorkflowFileReference> WORKFLOWS = new ListParameter<WorkflowFileReference>(
+                "workflows") {
+            @Override
+            TypeReference<List<WorkflowFileReference>> getType() {
+                return new TypeReference<List<WorkflowFileReference>>() {
+                };
+            }
+        };
+        /**
+         * code_scanning_tools parameter
+         */
+        public static final ListParameter<CodeScanningTool> CODE_SCANNING_TOOLS = new ListParameter<CodeScanningTool>(
+                "code_scanning_tools") {
+            @Override
+            TypeReference<List<CodeScanningTool>> getType() {
+                return new TypeReference<List<CodeScanningTool>>() {
+                };
+            }
+        };
+    }
+
+    /**
+     * Basic parameter for a ruleset.
+     *
+     * @param <T>
+     *            the type of the parameter
+     */
+    public abstract static class Parameter<T> implements Function<JsonNode, T> {
+
+        private final static ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
Definitely do not create a new object mapper for this class.  Reuse the one in GitHubClient, it is configured to correctly handle json returned from GitHub.